##  Guide to get curser coordinates for AutoZOOM to work

This is a small guide to get cursor coordinates appropriate for operating zoom from python script


Some technical notes:

- main script: [https://github.com/itsvenu/DataScience-Projects/blob/master/AutoZOOM/AutoZOOM_v0.03.py](https://github.com/itsvenu/DataScience-Projects/blob/master/AutoZOOM/AutoZOOM_v0.03.py)

- 13 inch display (MacBook pro)
- chrome browser
- I did not grant any permissions to zoom app installed in my laptop
- Video is off by default in my zoom app

- I did not include `close zoom` option in the script, just to not look like completely automated




If you have different screen size and different zoom settings, following guide will help you automate the zoom joining.

In [1]:
# required module

import pyautogui as pyg # required module

### step-1

As soon as you enter zoom meeting link in chrome browser, it will give a pop-up window like following

![popup](zoom_popup.png)

Keep the curser on **Open zoom.us.app** and run the following command

In [3]:
pyg.position()

Point(x=814, y=246)

In my laptop, the pop-up is approximately at Point(x=814, y=246). Use these coordinates in line-45 of the main script [here](https://github.com/itsvenu/DataScience-Projects/blob/master/AutoZOOM/AutoZOOM_v0.03.py)

### Step-2

After clicking **Open zoom.us.app**, it will open the zoom app that is installed in your laptop. It will look something like the following

![app](zoom_app.png)

By default, if there is no one sharing a screen/presenting slides, the app will open in a small window. We need to maximize the zoom app window. Keep the curser on maximize circle of the zoom app and again run the above command

In [4]:
pyg.position()

Point(x=194, y=40)

In my laptop, this is close to Point(x=194, y=40). Now the app window is maximized. Add these coordinates in line-47 of the main script.

### Step-3

Now, we need to find the coordinates for the mute button in maximized zoom app. Repeat the same procedure as above. Keep the cursor on mute button on zoom app and run the `pyg.position()` command. In my laptop, this gave me Point(x=50, y=776). Add these coordinates to the line-49 of the main script.